test/includes.spec.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const _ = require('lodash')
+const Promise = require('bluebird')
+const Joi = require('joi')
+const Hapi = require('hapi')
+
+let server, buildServer, destroyServer, hh;
+
+const schema = {
+    type: 'brands',
+    attributes: {
+        code: Joi.string().min(2).max(10),
+        description: Joi.string(),
+        year: Joi.number()
+    }
+};
+
+const data = {
+    attributes: {
+        code: 'MF',
+        description: 'Massey Furgeson',
+        year: 2007
+    }
+};
+
+
+//TODO just done the validation, actual includes is remaining
```

Comment:
vocabulary was mixed a bit here perhaps

Includes is the clause which allows to include related resources and return a composite data graph : http://jsonapi.org/format/#fetching-includes

The behaviour which seems to have been implemented and tested here is sparse fields : http://jsonapi.org/format/#document-resource-object-fields